app/models/concerns/offeror.rb

```diff
@@ -30,20 +30,26 @@ def import(file, comment)
   def remove_unclaimed_codes(offeror_codes)
     @unclaimed_codes = offeror_codes.where(user_id: nil)
     @num = self.unclaimCodes
-    @date = Time.now.to_formatted_s(:long_ordinal)
-    @contents = "There are #{@num} unclaimed codes, removed on #{@date}\r\n\r\n"
     @unclaimed_codes.each do |code|
-      @contents = "#{@contents}#{code.code}\r\n"
       code.destroy
     end
     
     comment = "Codes were removed"
     update_history(comment)
-
     self.update(unclaimCodes: 0, removedCodes: self.removedCodes + @num)
-    return @contents
   end
   
+  def download_codes(offeror_codes)
+    @unclaimed_codes = offeror_codes.where(user_id: nil)
+    @num = self.unclaimCodes
```

Comment:
Yes, perhaps, as will be the function of downloading used codes. I kept the shorter name associating it with unclaimed codes by default and to not to make it excessively long, but as you've paid attention to this, let's change to more evident.